common.c

```diff
@@ -65,6 +65,88 @@ int g_enable_tracing = 1;
 struct descriptor_info **g_descriptor_list;
 unsigned int g_descriptor_list_size;
 
+// special strings to be displayed with ANSI colors
+struct str_colors {
+	char ch;
+	const char *color_str;
+};
+
+static struct str_colors g_str_colors[] = {
+	{ '\n', VAR "\\n" RST },
+	{ '\r', VAR "\\r" RST },
+	{ '\t', VAR "\\t" RST },
+	{ '\0', NULL }
+};
+
+static void
+trace_printf_str(const char *str)
+{
+	char *str_print;
+
+	// get original function pointers
+	rtr_malloc_t real_malloc = RETRACE_GET_REAL(malloc);
```

Comment:
Please don't mix declarations and function calls. I.e. it should look something like:

char *str_print;
rtr_malloc_t real_malloc;
rtr_free_t real_free;
....

real_malloc = RETRACE_GET_REAL(malloc);
real_free = RETRACE_GET_REAL(free);